__tests__/integration.test.js

```diff
@@ -0,0 +1,39 @@
+const app = require("../db/app");
+const db = require("../db/connection");
+const data = require("../db/data/test-data/index");
+const seed = require("../db/seeds/seed");
+const request = require("supertest");
+
+beforeEach(() => {
+  return seed(data);
+});
+
+afterAll(() => {
+  return db.end();
+});
+
+describe("Any incorrect URL", () => {
+  test("ANY 404: For incorrect URL input, responds with a bad request error", () => {
+    return request(app)
+      .get("/api/wrong_URL")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.message).toBe("URL not found");
+      });
+  });
+});
+
+describe("/api/topics", () => {
+  test("GET 200 - Responds with all topics.", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.length).toBe(3);
+        body.forEach((topic) => {
+          expect(typeof topic.description).toBe("string");
```

Comment:
one way to do this that I think looks a bit nicer is using .toMatchObject or .objectContaining (both can still be used within a loop), but either way great job!